### PR TITLE
added "allSettled" functionality

### PR DESCRIPTION
**Pull request by goodwin64**
| Q                        | A  |
| ------------------------ | ---  |
| Fixed Issues?            | https://github.com/redux-saga/redux-saga/issues/1800 |
| Patch: Bug Fix?          |  No  |
| Major: Breaking Change?  |  No  |
| Minor: New Feature?      |  Yes  |
| Tests Added + Pass?      | Yes, but prettier warns about `console.warn` |
| Any Dependency Changes?  | No  |

Added the `allSettled` functionality with the same API as in Promise.

### CODE REVIEW

File: packages/core/src/internal/io.js [hunk 1]
```diff
@@ -279,3 +279,16 @@ export function setContext(props) {
 }
 
 export const delay = call.bind(null, delayP)
+
+export const allSettled = effects =>
+  all(
+    effects.map(effect =>
```
Comment (Andarist) [inline packages/core/src/internal/io.js:285]: Currently, the redux-saga's code assumes ES5 environment and thus we cannot use generator syntax in our source code. All other helpers are implemented using the custom `fsmiterator` approach which looks like this: https://github.com/redux-saga/redux-saga/blob/8f8d9e1d8bb957e9d3253ea250b317fde4fafb09/packages/core/src/internal/sagaHelpers/throttle.js#L5 

I understand that this is somewhat quirky so if you don't want to look into rewriting this to accommodate our needs I will try to find the time to do it later myself - just let me know.

Comment (goodwin64) [reply packages/core/src/internal/io.js:285]: Thanks for the note. I need some time to read about the `fsm-iterator` approach to understand its' internals. I'll write here soon whether I can adjust this PR to this approach.

Comment (goodwin64) [reply packages/core/src/internal/io.js:285]: No ideas for now. I'd be grateful if you will take it once to work 🙏😇

Comment (Andarist) [reply packages/core/src/internal/io.js:285]: Sure - I will handle this sometime later. If you still are interested in understanding `fsmIterator` a little bit better I've dug up an old comment of mine which from what I remember shed some light on this one: https://github.com/redux-saga/redux-saga/issues/570#issuecomment-256717377

**Pull request closed without merging**